### PR TITLE
Github actions MacOS CI 

### PR DESCRIPTION
**Pull request by r0mai**
Had to fix two test failures:
* A bug in boost wave prevented parsing some system headers due to a bug in the `__has_include` grammar. For this I cherry picked some parts of a boost::wave fix: https://github.com/boostorg/wave/pull/144 (not yet released in an official boost.wave release). I also enabled `__has_include` parsing by default. Normally it is only enabled on C++17 and above.
* Another test failure was caused by `/proc` being used as read-only directory. `/proc` doesn't exist on macOS, so a simple `chmod` based approach was used.

### CODE REVIEW

File: lib/system_test/src/read_only_path.cpp [hunk 1]
```diff
@@ -27,11 +27,13 @@ namespace metashell
 {
   namespace system_test
   {
-#ifdef _WIN32
     namespace
     {
+#if defined(_WIN32) || defined(__APPLE__)
```
Comment (sabel83) [inline lib/system_test/src/read_only_path.cpp]: Why is this needed on `__APPLE__`?

Comment (r0mai) [reply lib/system_test/src/read_only_path.cpp]: Thanks for noticing, this was left in accidentally
